### PR TITLE
DS-7989 - PDS 2.0 testcase refactor - DeletePDSPods

### PR DESCRIPTION
**Pull request by aravindranath14**
delete pds pods and validate if its coming back online and dataServices are not affected

Successful Jenkins Run:
https://jenkins.pwx.dev.purestorage.com/job/PDS/job/pds2.0-byoc-branch-build-test/431

### CODE REVIEW

File: drivers/unifiedPlatform/pdsLibs/dsLib.go [hunk 1]
```diff
@@ -203,3 +205,54 @@ func DeleteAllDeployments(projectId string) error {
 	log.Infof("Total number of deployments Deleted [%d]", numberOfDeploymentsDeleted)
 	return nil
 }
+
+// GetPods returns the list of pods in namespace
+func GetPods(namespace string) (*corev1.PodList, error) {
```
Comment (atrivedi-px) [inline drivers/unifiedPlatform/pdsLibs/dsLib.go:210]: Could we move this method to k8utils.go?

File: drivers/unifiedPlatform/pdsLibs/dsLib.go [hunk 2]
```diff
@@ -203,3 +205,54 @@ func DeleteAllDeployments(projectId string) error {
...
+	k8sOps := k8sCore
+	podList, err := k8sOps.GetPods(namespace, nil)
+	if err != nil {
+		return nil, err
+	}
+	return podList, err
+}
+
+// DeleteDeploymentPods deletes the given pods
+func DeletePods(podList []corev1.Pod) error {
```
Comment (atrivedi-px) [inline drivers/unifiedPlatform/pdsLibs/dsLib.go:220]: Could we move this method to k8utils.go?

File: drivers/unifiedPlatform/pdsLibs/dsLib.go [hunk 3]
```diff
@@ -203,3 +205,54 @@ func DeleteAllDeployments(projectId string) error {
...
+	err := k8sCore.DeletePods(podList, true)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// ValidatePods returns err if pods are not up
+func ValidatePods(namespace string, podName string) error {
```
Comment (atrivedi-px) [inline drivers/unifiedPlatform/pdsLibs/dsLib.go:229]: Could we move this method to k8utils.go?

**Review by atrivedi-px — APPROVED** [review]
LGTM

**Review by Madanagopal19 — APPROVED** [review]

**Pull request merged**